File: src/hasoftware-library/src/hasoftware/cdef/CDEFAction.java
package hasoftware.cdef;

public class CDEFAction {

    public static final int None = 0;
    public static final int Create = 1;
    public static final int Delete = 2;
    public static final int List = 3;
    public static final int Update = 4;

    private static String[] Actions = new String[]{
        "None",
        "Create",
        "Delete",
        "List",
        "Update"
    };

    public static String getActionStr(int action) {
        return Actions[action];
    }
}

File: src.net/hasoftware-library/Cdef/CdefAction.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hasoftware.Cdef
{
    public enum CdefAction
    {
        [Description("None")]
        None = 0,
        [Description("Create")]
        Create = 1,
        [Description("Delete")]
        Delete = 2,
        [Description("List")]
        List = 3,
        [Description("Update")]
        Update = 4
    }
}
